xbmc/network/Network.cpp

```diff
@@ -159,19 +159,49 @@ int CNetwork::ParseHex(char *str, unsigned char *addr)
    return len;
 }
 
-std::string CNetwork::GetHostName(void)
+bool CNetwork::GetHostName(std::string& hostname)
 {
   char hostName[128];
   if (gethostname(hostName, sizeof(hostName)))
-    return "unknown";
+    return false;
 
-  std::string hostStr;
 #ifdef TARGET_WINDOWS
+  std::string hostStr;
   g_charsetConverter.systemToUtf8(hostName, hostStr);
+  hostname = hostStr;
 #else
-  hostStr = hostName;
+  hostname = hostName;
 #endif
-  return hostStr;
+  return true;
+}
+
+bool CNetwork::IsLocalHost(const std::string& hostname)
+{
+  if (hostname.empty())
+    return false;
+
+  if ((hostname == "127.0.0.1")
```

Comment:
In this case though it'll be the user who enters the address that we want to check here. I doubt anyone would enter e.g. 127.2.18.1, but since this method is generic it should probably handle all cases so we don't get burned in the future.